includes/data-port/class-sensei-data-port-job

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * File containing the Sensei_Data_Port_Job class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Superclass of data import/export jobs. It provides basic functionality like logging, maintaining state, cleanup and
+ * running the data port for each level.
+ */
+abstract class Sensei_Data_Port_Job implements Sensei_Background_Job_Interface {
+	private const OPTION_PREFIX = 'sensei-tools-';
+
+	/**
+	 * An array which holds the results of the data port job and populated in subclasses.
+	 *
+	 * @var array
+	 */
+	protected $results;
+
+	/**
+	 * Unique id for the job.
+	 *
+	 * @var string
+	 */
+	private $job_id;
+
+	/**
+	 * An array containing log entries (e.g. errors). It has the following format:
+	 * [
+	 *    name => The name of the entity this entry applies to.
+	 *    id   => The id of the entity this entry applies to.
+	 *    msg  => The log message
+	 * ]
+	 *
+	 * @var array
+	 */
+	private $logs;
+
+	/**
+	 * True if the job is completed.
+	 * @var bool
+	 */
+	private $is_completed;
+
+	/**
+	 * True if the internal state has changed and needs to be stored.
+	 *
+	 * @var bool
+	 */
+	private $has_changed;
+
+	/**
+	 * An array which contains the state for each subjob, i.e. questions, lessons and courses.
+	 *
+	 * @var array
+	 */
+	private $sub_jobs_state;
+
+	/**
+	 * Estimate of completion percentage.
+	 *
+	 * @var float
+	 */
+	private $percentage;
+
+	/**
+	 * Sensei_Data_Port_Job constructor.
+	 *
+	 * @param string $job_id   Unique job id.
+	 * @param array $sub_jobs  A list of subjobs for this job.
```

Comment:
I think this should probably be an abstract function in the sub-class. `function get_tasks(){}` maybe?